### DISCUSSION TITLE
Want to output the matrix of coefficients in the terminal

### DISCUSSION THREAD

**Discussion by Raven-pro** [idaholab/moose#22292] [Q&A General] [answered]
Hello, developers!

I am trying to know if my codes or physical equations are correct and I wanna check my matrix of coefficients, and the Jacobian matrix.

Is there any method or option to output the matrix of coefficients in the terminal? The size of the problem is not huge, just want to have a little check of my codes so I can debug.

Thank you!

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can use `-ksp_view_mat` to view the matrix. You can pass it on the command line or add it to the `-petsc_options` in the Executioner. 

You should look at the petsc manual for more info on this 

Guillaume

**Reply by Raven-pro** [reply]
Thanks! I will look into it.

**Reply by ghost** [reply]
Hi @89018580 ,

I am trying to do something similar. I am looking at outputting the coefficient matrices to solve for eigenvalues in MATLAB. Were you able to output coefficient matrices for your case? I did go through the PETSc manual, and found an option `-mat_view` which outputs a matrix. However, not being familiar with the PETSc system, I am not able to identify what part the matrix belongs to.

**Reply by lindsayad** [reply]
`-ksp_view_pmat` is more reliable as it covers both NEWTON and PJFNK solve types

`-mat_view` will print any/all matrices right after they have been assembled. In the case of MOOSE though there is only one matrix. For NEWTON this matrix is used both for the Jacobian and preconditioner. For PJFNK it is used for the preconditioner (the Jacobian is approximated matrix-free)

**Reply by ghost** [reply]
Is there a way to know which part of the matrix belongs to which kernel? (corresponding term in the equation). 

The problem I am trying to solve is attached below. What I am trying to output, is coefficient matrices A, B and C, as highlighted in the attached description and then use those matrices to solve for the quadratic eigenvalue. Any help would be greatly appreciated.

[Reference .pdf](https://github.com/idaholab/moose/files/11568470/Reference.pdf)

**Reply by lindsayad** [reply]
Yea it's not clear to me what A, B, and C should be, but you can save different kernels to different matrices using the tagging system. See https://mooseframework.inl.gov/framework_development/tagging.html and the input files in the `moose/test/tests/tag` directory

**Reply by ghost** [reply]
Thanks! It worked.

**Reply by lindsayad** [reply]
awesome, great to hear!
